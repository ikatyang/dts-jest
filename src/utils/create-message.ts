import { indent } from './indent';

export const create_message = (
  description: string,
  data: string[],
  footer?: string,
) => {
  const content = data
    .map(line_content => indent(line_content, 2))
    .join('\n')
    .replace(/ +$/gm, '');

  return footer === undefined
    ? `${description}\n\n${content}`
    : `${description}\n\n${content}\n\n${footer}`;
};
