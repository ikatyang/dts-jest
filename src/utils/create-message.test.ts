import { create_message } from './create-message';

it('should return correctly', () => {
  expect(
    create_message('my description', ['data 1', 'data 2', 'data 3']),
  ).toMatchSnapshot();
});
