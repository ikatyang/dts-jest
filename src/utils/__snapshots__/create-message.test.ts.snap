// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return correctly 1`] = `
"my description

  data 1
  data 2
  data 3"
`;
